hbase-server/src/main/java/org/apache/hadoop/hbase/master/AssignmentManager.java

```diff
@@ -3293,7 +3295,7 @@ boolean waitUntilNoRegionsInTransition(final long timeout)
         HRegionInfo regionInfo = hrl.getRegionInfo();
         if (regionInfo == null) continue;
         int replicaId = regionInfo.getReplicaId();
-        State state = RegionStateStore.getRegionState(result, replicaId);
+        State state = RegionStateStore.getRegionState(result, replicaId, server.getConfiguration());
```

Comment:
Can we just pass through useZk as a boolean instead of the whole configuration object? Not a big deal, but the former seems a bit cleaner.